test/test.js

```diff
@@ -29,6 +30,32 @@ describe("Send Endpoint", function () {
     };
   });
 
+  it("Handle when a conntection error is thrown", function (done) { 
+    const data = () => {
+     return new Error('ETIMEDOUT')
```

Comment:
`.catch(({ response, result }) => { that._handleResponse.call(that, result, response, callback); })`
Will execute the provided callback for any thrown errors (400 and 500) as well as including network error because `Sendwithus.prototype._handleResponse` now handles for throws outside of the default Error instance if they have a message and stack trace. Furthermore, responses which are not error but are not considered 200 group responses are also returned (100, and 300)

Reply:
Mocking a isnt ideal because the failures could be due to many possible reasons outside of the scope this change. Depending on the the version of the Nodejs running and the OS architecture this does make it hard to determine a proper test that would validate correctness. I will leave some reference help links for future reference if this becomes a notable issue in the future:
http://willbryant.net/overriding_the_default_linux_kernel_20_second_tcp_socket_connect_timeout
https://medium.com/hk01-tech/running-eks-in-production-for-2-years-the-kubernetes-journey-at-hk01-68130e603d76
https://www.joyent.com/node-js/production/design/errors